File: helpdeskspringboot/src/main/java/com/ez/payload/SlaReportHeader.java
package com.ez.payload;

public interface SlaReportHeader {

    // "numOfTickets" column.
    // number of tickets between fromDate toDate, and based on team filter.
    long getNumOfTickets();

    // "numOfOntimeTickets" column.
    // number of ontime tickets between fromDate toDate, and based on team filter.
    long getNumOfOntimeTickets();

    // "numOfLatedTickets" column.
    // number of lated tickets between fromDate toDate, and based on team filter.
    long getNumOfLatedTickets();

    // "SlaPercent" column(SLA percentage) = (numOfOntimeTickets/numOfTickets)
    float getSlaPercent();
}

File: helpdeskangular/src/app/payload/SlaReportHeader.ts
export class SlaReportHeader {

    constructor(

        // total of tickets from fromDate to toDate of each teamid
        public numOfTickets: number,

        // total of ontime tickets for each teamid
        public numOfOntimeTickets: number,

        // total of lated tickets for each teamid
        public numOfLatedTickets: number,

        // sla percentage
        public slaPercent: string,



    ) {

    }

}
